it/uk/gov/hmrc/agentsubscriptionfrontend/support/EndpointBehaviours.scala

```diff
@@ -62,4 +69,114 @@ trait EndpointBehaviours {
+      (request, result)
+    }
+
+    "include absolute continue URL" in {
+      val url = "http://localhost"
+      val (request, result) = doRequestWithContinueUrl(url)
+      assertContinueUrlKept(request, result, url)
+    }
+
+    "include relative continue URL" in {
+      val url = "/foo"
+      val (request, result) = doRequestWithContinueUrl(url)
+      assertContinueUrlKept(request, result, url)
+    }
+
+    "include continue URL if it's the absolute www.tax.service.gov.uk continue url" in {
+      val url = "http://www.tax.service.gov.uk/foo/bar?some=true"
+      val (request, result) = doRequestWithContinueUrl(url)
+      assertContinueUrlKept(request, result, url)
+    }
+
+    "include continue URL if it's whitelisted" in {
+      val url = "http://www.foo.com/bar?some=false"
+      val (request, result) = doRequestWithContinueUrl(url)
+      assertContinueUrlKept(request, result, url)
+    }
+
+    "not include a continue URL if it's not whitelisted" in {
+      val url = "http://www.foo.org/bar?some=false"
+      val (request, result) = doRequestWithContinueUrl(url)
+      assertContinueUrlNotKept(request, result, Some(url))
+    }
+
+    "not include a continue URL if it contains an invalid character" in {
+      val url = "http://www@foo.com"
+      val (request, result) = doRequestWithContinueUrl(url)
+      assertContinueUrlNotKept(request, result, Some(url))
+    }
+
+    "not include a continue URL if it's not provided" in {
+      val request = FakeRequest("GET", "/").withSession(sessionKeys: _*)
+      val result = await(action(request))
+      assertContinueUrlNotKept(request, result, None)
+    }
+  }
+
+  protected def anEndpointTakingContinueUrlAndRedirectingWithIt(action: PlayRequest): Unit = {
+    def continueParamRegex(expectedContinueUrl: String) = s"[\\?&]continue=${urlencoded(expectedContinueUrl)}".r
+
+    def checkRedirectHasContinueUrl(request: Request[AnyContent], result: Result, expectedContinueUrl: String) = {
+      status(result) shouldBe 303
+      result.header.headers(LOCATION) should include regex continueParamRegex(expectedContinueUrl)
+    }
+
+    def checkRedirectHasNoContinueUrl(request: Request[AnyContent], result: Result, expectedContinueUrl: Option[String]) = {
+      status(result) shouldBe 303
+      result.header.headers(LOCATION) should not include "continue="
+    }
+
+    aPageTakingContinueUrl(action, Seq(), checkRedirectHasContinueUrl, checkRedirectHasNoContinueUrl)
+  }
+
+  protected def aPageTakingContinueUrlAndContainingItAsALink(action: PlayRequest): Unit = {
+    def checkResultHasLinkWithContinueUrl(request: Request[AnyContent], result: Result, expectedContinueUrl: String) = {
+      status(result) shouldBe 200
+      bodyOf(result) should include regex(s""".*<a .*href=.*\\?continue=${urlencoded(expectedContinueUrl)}.*</a>""".r)
+    }
+
+    def checkResultHasNoLinkWithContinueUrl(request: Request[AnyContent], result: Result, expectedContinueUrl: Option[String]) = {
```

Comment:
noLinkWithContinueUrl? not sure whether I am confused by the name of the function or by the code of it.
if there is not supposed to be a continue URL then there should never be a "continue=" in the url